### DISCUSSION TITLE
Filter with HAVING is ignored

### DISCUSSION THREAD

**Discussion by madsem** [filamentphp/filament#5422] [Help]
Hey,

trying to create a Filter with a `having` clause, but it simply gets ignored and is not appended to the query.
Is there a way to make a filter like this work?

```php
Filter::make('revenue_greater_than')
                ->form([
                    Forms\Components\TextInput::make('greater_than')
                        ->label('RPL Greater Than'),
                ])
                ->query(function (Builder $query, array $data): Builder {
                    return $query
                        ->when(
                            $data['greater_than'],
                            function (Builder $query, $dollar) {

                                $money = Money::of($dollar, config('settings.currency'))->getMinorAmount()->toInt();

                                return $query->havingRaw("SUM(revenue) / COUNT(sold_at) >= {$money}");
                            },
                        );
                }),
```

**Comment by danharrin** [comment]
`having()` is for groups, I think you can just use `whereRaw()`?

**Reply by madsem** [reply]
@danharrin  thanks! But sadly no, because the table query is also grouped:

```php
protected function getTableQuery(): Builder
    {
        return Lead::query()
            ->selectRaw('SUM(revenue) AS revenue, state, state_code')
            ->selectRaw('COUNT(lead_campaign_identifier) AS total_leads')
            ->selectRaw('COUNT(sold_at) AS total_sold')
            ->groupBy('state', 'state_code');
    }
```

So it seems thhis is only possible by using `having`? But this isn't supported by the filters in filament as far as I can gather?

**Reply by danharrin** [reply]
The table builder relies heavily on Eloquent models. By adding selects and groups to the main query, that breaks model results. You need to use a subquery and `whereIn()` to achieve this sort of thing in a filter

**Reply by underdpt** [reply]
Hello,

I'm also stuck with this. I'm sorry I disagree with @danharrin , `HAVING` clauses can be used without grouping, and thus won't break the eloquent result.

My use case is a query with a column that's the result of some heavy sql filtering and aggregation. I want to filter by that column, and also have that column on the results. Using `having` i have to calculate that query only once, but using `where` i would have to run the subquery twice, one for the filtering and one to attach the column to the models. That's a hit on the app's performance I would like to avoid.

Light example on #6767
